samples/SwiftDemoApp/SwiftDemoApp/Samples.swift

```diff
@@ -13,25 +13,25 @@
  * limitations under the License.
  */
 
-import Foundation
+import UIKit
 
 class Samples {
-  class func loadSamples() -> [Any] {
-    return [ newDemo(controller: ClusteringViewController.self,
+    class func loadSamples() -> [[String: Any]] {
+    return [ newDemo(controller: ClusteringViewController(),
```

Comment:
this code potentially has the leaking memory issue, please be aware that this.

Reply:
What is the reason behind this? It's always preferred to use specific data types rather than, to make the return type developer friendly.

Reply:
The main reason is the `MasterViewController` declares the samples variable which is the strong reference type. 
https://github.com/googlemaps/google-maps-ios-utils/blob/024fbf0be909ed5e95db56e3ec9d544978a9bcb5/samples/SwiftDemoApp/SwiftDemoApp/MasterViewController.swift#L20
the MasterViewController use `Samples.LoadSamples()` and assign to `samples` variable. 
https://github.com/googlemaps/google-maps-ios-utils/blob/024fbf0be909ed5e95db56e3ec9d544978a9bcb5/samples/SwiftDemoApp/SwiftDemoApp/MasterViewController.swift#L26
So every object in sample array keeps in memory until MasterViewController deallocated. 

for instance, you instantiate the `ClusteringViewController()` object in the dictionary, then assign it to the `samples` variable. So `ClusteringViewController` object keeps in memory until the master view controller object deallocated from memory. 

this causes the memory leak and does not optimize code.

Reply:
If we can gracefully handle the scenario when `MasterViewController` deallocates[TBD], I don't feel any leaks will be there.

But that's how only it's written previously also `ClusteringViewController()` and others were in the memory until it deallocates. I am trying to understand, what extra I have added so it will create a memory leak?

Do you think I have changed the code from `ClusteringViewController.self` to `ClusteringViewController()` will create a leak?

Moreover, I have not made any optimization in the code just changed the return type to be more clear.

Hope you got it.

Reply:
Yes, The point is, we stored Type of `AnyClass` in memory before. but the changes store an instance of the ViewController object.  If you see in the line `viewController.init()` in MasterViewController, you understand that we create a new instance when the cell did select.
Note: After an instance of view controller pop from the navigation view controller. ARC automobility deallocate from memory.

Reply:
this makes sense now. `viewController.init()`, updating PR. Updated the PR. Thanks @farshadmb 👍

Reply:
Good callout @farshadmb